tram-one.js

```diff
@@ -58,66 +103,144 @@ class Tram {
     })
 
+    // if webStorage is defined, wire it into hover-engine with hover-battery
+    // (effectively, set up the initial store values and add appropriate listeners to write to webStorage)
     if (this.webStorage) {
       this.engine.addActions(battery(this.webStorage).actions)
       this.engine.addListener(battery(this.webStorage).listener)
     }
 
+    // wire up urlListener so that we remount whenever the url changes
     urlListener(() => {
       this.mount(selector, pathName)
     })
 
+    // trigger an initial mount
     this.mount(selector, pathName)
 
     return this
   }
 
+  /**
+   * internal method for building and updating / creating the app
+   * - do not call this to start your app
+   * - for complete documentation, please refer to http://tram-one.io/#app-mount
+   *
+   * @param {*} selector
+   * @param {string} [pathName]
+   * @param {object} store
+   * @param {object} actions
+   */
   mount(selector, pathName, store, actions) {
     assert.ok(selector !== undefined, 'Tram-One: selector should be a DOM element or CSS selection string')
 
+    // if the selector is a string, try to find the element,
+    // otherwise it's probably DOM that we should write directly to
     const target = (typeof selector) === 'string' ? document.querySelector(selector) : selector
     if (target === null) {
       console.warn('Tram-One: could not find target, is the element on the page yet?')
     }
+
+    // build a div to render the app on
+    // (if it doesn't exist as a child of the selector, create one first)
     if (!target.firstElementChild) {
       const targetChild = document.createElement('div')
       target.appendChild(targetChild)
     }
     const targetChild = target.firstElementChild
 
+    // use the pathname provided, or pull the url from the window object
     const routePath = pathName || window.location.href.replace(window.location.origin, '')
+
+    // collect all the DOM events that we should be keeping track of.
+    // these events are provided by belit and consumed by tatermorph.
+    // (events are strange and actually aren't natively stored on Nodes,
+    // but we have to keep track of them so we can know to add or
+    // remove them between renders)
     const getEvents = (newNode, oldNode) => {
       return [].concat(newNode.events).concat(oldNode.events)
     }
+
+    // update our child element with a new version of the app
+    // tatermorph knows how to intelligently trigger only diffs that matter
     morph(targetChild, this.toNode(routePath, store, actions), getEvents)
 
     return this
   }
 
-  toNode(pathName, state, actions) {
+  /**
+   * generate a DOM Node Tree based on a path and a store
+   * - for complete documentation, please refer to http://tram-one.io/#app-toNode
+   *
+   * @param {string} pathName
+   * @param {object} [store]
+   * @param {object} [actions]
+   *
+   * @return {HTMLDivElement}
+   */
+  toNode(pathName, store, actions) {
```

Comment:
Thoughts of using optional parameters here? Obviously this wouldn't be equivalent, but I think if someone wants to pass in a false-y store and actions you shouldn't be overriding them.

Reply:
Actually I would love to changes these all up to use parameters deconstruction... However I don't know if I could do type checking then... Or which is better

Reply:
You could still do optional parameters with objects. And probably still do the asserts your doing above.
```js
toNode({pathName, store=this.engine.store, actions=this.engine.actions}) {
```

Or you could always do a typeof check for objects and support objects and individual parameters